AppCenter/AppCenter/Internals/DelegateForwarder/MSDelegateForwarder.h

```diff
@@ -0,0 +1,72 @@
+#import <Foundation/Foundation.h>
+@protocol MSCustomDelegate;
+
+@interface MSDelegateForwarder : NSObject
+
+/**
+ * Enable/Disable Application forwarding.
+ */
+@property(nonatomic) BOOL enabled;
+
+/**
+ * Hash table containing all the delegates as weak references.
+ */
+@property(nonatomic) NSHashTable<id<MSCustomDelegate>> *delegates;
+
+/**
+ * Hold the original setDelegate implementation.
+ */
+@property(nonatomic) IMP originalSetDelegateImp;
+
+/**
+ * Keep track of the original delegate's method implementations.
+ */
+@property(nonatomic, readonly) NSMutableDictionary<NSString *, NSValue *> *originalImplementations;
```

Comment:
It's true, it's also probably faster, though I'll just add a TODO since that was already the case before. Also we manipulate the string directly [here ](https://github.com/Microsoft/AppCenter-SDK-Apple/pull/1191/files/2821e1e8e10b647653fb16b5fab01b11ba98b78b#diff-033759a6401f4841724c1b1c53ede8b9R112) but we can deal with it even with a value.